unlock-app/src/components/interface/locks/Settings/forms/CreditCardCustomPrice.tsx

```diff
@@ -0,0 +1,168 @@
+    lockAddress,
+    network,
+    currencyContractAddress,
+    amount: parseFloat(keyPrice),
+  })
+
+  const hasPriceConversion = fiatPricing?.usd?.amount
+
+  const onSaveCreditCardPrice = async ({
+    creditCardPrice,
+  }: CreditCardFormSchema) => {
+    const price = creditCardPrice
+      ? parseFloat(`${creditCardPrice}`) * 100 // save price in basis points
+      : null
+
+    const savePricePromise = saveSettingMutation({
+      lockAddress,
+      network,
+      creditCardPrice: price,
+    })
+
+    await ToastHelper.promise(savePricePromise, {
+      loading: 'Updating price...',
+      success: 'Price updated.',
+      error: 'There is some issue updating the price.',
+    })
+  }
+
+  const onSubmit = (fields: CreditCardFormSchema) => {
+    onSaveCreditCardPrice(fields)
+  }
+
+  if (isLoading) {
+    return (
+      <Placeholder.Root>
+        <Placeholder.Card />
+      </Placeholder.Root>
+    )
+  }
+
+  const saveDisabled = isSaveLockSettingLoading || disabled
+
+  const symbol = lock?.currencySymbol
+
+  return (
+    <div className="grid gap-2">
+      <SettingCardDetail
+        title="Set fixed price in $USD"
+        description={
+          <span className="text-sm">
+            {`If you set a custom price for card payments, we will use that one (+ fees) instead of converting the price of ${symbol} to USD.`}
+          </span>
+        }
+      />
+      <div className="grid grid-cols-1 gap-1 p-4 bg-gray-100 rounded-lg">
+        <form
+          onSubmit={handleSubmit(onSubmit)}
+          className="grid items-center gap-3"
+        >
+          <ToggleSwitch
+            title="Use fixed price"
+            enabled={useCustomPrice}
+            setEnabled={setUseCustomPrice}
+            onChange={(enabled) => {
+              if (!enabled) {
+                setValue('creditCardPrice', null, {
+                  shouldValidate: true,
+                  shouldDirty: true,
+                })
+              }
+            }}
+          />
+          {useCustomPrice && (
+            <Input
+              type="number"
+              step="any"
+              disabled={disabled}
+              description="Set a fixed price in USD that is charged for card payments."
+              error={errors?.creditCardPrice?.message}
+              {...register('creditCardPrice', {
+                required: true,
+                min: {
+                  value: CREDIT_CARD_MIN_USD_PRICE,
+                  message: `Price is too low for us to process credit cards. It needs to be at least ${parseFloat(
+                    `${CREDIT_CARD_MIN_USD_PRICE}`
+                  ).toFixed(2)}. `,
+                },
+              })}
+            />
+          )}
+          <div className="w-full md:w-1/3">
+            <Button className="w-full" size="small" disabled={saveDisabled}>
+              Apply
+            </Button>
+          </div>
+        </form>
+      </div>
+      {!hasPriceConversion && (
```

Comment:
Please enable the toggle by default if indeed `hasPriceConversion` is falsy and only show the message below if the user has not set a price for cards.